release/ray_release/file_manager/job_file_manager.py

```diff
@@ -41,20 +54,24 @@ def _generate_tmp_s3_path(self):
         location = f"tmp/{generate_tmp_s3_path()}"
         return location
 
-    def download_from_s3(
+    def download_from_cloud(
         self, key: str, target: str, delete_after_download: bool = False
     ):
-        # s3 -> local target
-        self._run_with_retry(
-            lambda: self.s3_client.download_file(
-                Bucket=self.bucket,
-                Key=key,
-                Filename=target,
+        if self.cloud_storage_provider == "s3":
+            self._run_with_retry(
+                lambda: self.s3_client.download_file(
+                    Bucket=self.bucket,
+                    Key=key,
+                    Filename=target,
+                )
             )
-        )
+        if self.cloud_storage_provider == "gs":
+            bucket = self.gs_client.bucket(self.bucket)
+            blob = bucket.blob(key)
+            self._run_with_retry(lambda: blob.download_to_filename(target))
 
         if delete_after_download:
-            self.delete(target)
+            self.delete(key)
```

Comment:
this is changed from deleting `target` to `key`, is this an intended change?

the `delete()` function's arg still says `target`.